src/gluonts/dataset/tabular.py

```diff
@@ -0,0 +1,213 @@
+    for example, are stacked on top of each other rather than side-by-side. The
+    same is true for other dynamic or categorical features.
+
+    Parameters
+    ----------
+    data: pd.DataFrame
+        pd.DataFrame containing at least timestamp, target and item_id columns.
+    target: str or List[str]
+        Name of the column that contains the target time series.
+        For multivariate targets `target` is expecting a list of column names.
+    timestamp: str
+        Name of the column that contains the timestamp information.
+    item_id: str
+        Name of the column that, when grouped by, gives the different time series.
+    freq: str
+        Frequency of observations in the time series. Must be a valid pandas frequency.
+    feat_dynamic_real: List[str]
+        List of column names that contain dynamic real features.
+    feat_dynamic_cat: List[str]
+        List of column names that contain dynamic categorical features.
+    feat_static_real: List[str]
+        List of column names that contain static real features.
+    feat_static_cat: List[str]
+        List of column names that contain static categorical features.
+    past_feat_dynamic_real: List[str]
+        List of column names that contain dynamic real features only for the history.
+    prediction_length: int
+        For target and past dynamic features last `prediction_length` elements are
+        removed when iterating over the data set. This becomes important when the
+        predictor is called.
+    """
+
+    data: pd.DataFrame
+    target: Union[str, List[str]]
+    timestamp: str
+    item_id: str
+    freq: str
+    feat_dynamic_real: List[str] = field(default_factory=list)
+    feat_dynamic_cat: List[str] = field(default_factory=list)
+    feat_static_real: List[str] = field(default_factory=list)
+    feat_static_cat: List[str] = field(default_factory=list)
+    past_feat_dynamic_real: List[str] = field(default_factory=list)
+    prediction_length: int = 0
+
+    def __post_init__(self) -> None:
+        self.one_dim_target = not isinstance(self.target, list)
+        self.process = ProcessDataEntry(
+            self.freq, one_dim_target=self.one_dim_target
+        )
+
+    def _dataentry(self, item_id: str, df: pd.DataFrame) -> DataEntry:
+        check_timestamps(df.loc[:, self.timestamp].to_list(), freq=self.freq)
+        dataentry = dataframeTS_to_dataentry(
+            data=df,
+            target=self.target,
+            timestamp=self.timestamp,
+            feat_dynamic_real=self.feat_dynamic_real,
+            feat_dynamic_cat=self.feat_dynamic_cat,
+            feat_static_real=self.feat_static_real,
+            feat_static_cat=self.feat_static_cat,
+            past_feat_dynamic_real=self.past_feat_dynamic_real,
+        )
+        dataentry["item_id"] = item_id
+        return dataentry
+
+    def __iter__(self) -> Iterator[DataEntry]:
+        for item_id, df in self.data.groupby(self.item_id):
```

Comment:
How about this class is constructed directly with a collection of `DataFrame` objects? For example, a dictionary `{item_id: df}` (or a list in case one doesn't need or want to associate an `item_id` to each series). And then to handle the long format we add a thin wrapper that does the `groupby`?

Something like
* `DataFramesDataset` (note the `s`)
* `LongDataFrameDataset`

where essentially

```python
def LongDataFrameDataset(df, ..., item_id_column):
    return DataFramesDataset(list(df.groupby(item_id_column)))
```

Reply:
Done. DataFramesDataset supports both. `List` and `Dict` now.